pkg/csi/cinder/nodeserver.go

```diff
@@ -29,26 +29,37 @@ import (
 
 	"k8s.io/cloud-provider-openstack/pkg/csi/cinder/mount"
 	"k8s.io/cloud-provider-openstack/pkg/csi/cinder/openstack"
+	cpoerrors "k8s.io/cloud-provider-openstack/pkg/util/errors"
 	"k8s.io/cloud-provider-openstack/pkg/util/metadata"
 )
 
 type nodeServer struct {
 	Driver   *CinderDriver
 	Mount    mount.IMount
 	Metadata openstack.IMetadata
+	Cloud    openstack.IOpenStack
 }
 
 func (ns *nodeServer) NodePublishVolume(ctx context.Context, req *csi.NodePublishVolumeRequest) (*csi.NodePublishVolumeResponse, error) {
 	klog.V(4).Infof("NodePublishVolume: called with args %+v", *req)
 
+	volumeID := req.GetVolumeId()
 	source := req.GetStagingTargetPath()
 	targetPath := req.GetTargetPath()
 	volumeCapability := req.GetVolumeCapability()
 
-	if len(source) == 0 || len(targetPath) == 0 || volumeCapability == nil {
+	if len(volumeID) == 0 || len(source) == 0 || len(targetPath) == 0 || volumeCapability == nil {
```

Comment:
might be overkill, but seems like unpublish to provide detail info will help debug...

Reply:
agreed providing exactly what is missing is good for debugging, we can have a separate PR for that, as this PR addressing some other concern